pynxtools/dataconverter/helpers.py

```diff
@@ -456,15 +459,28 @@ def ensure_all_required_fields_exist(template, data, nxdl_root):
             opt_parent = check_for_optional_parent(path, nxdl_root)
             if opt_parent != "<<NOT_FOUND>>":
                 if does_group_exist(opt_parent, data) and not does_group_exist(renamed_path, data):
-                    raise ValueError(f"The required group, {path}, hasn't been supplied"
-                                     f" while its optional parent, {path}, is supplied.")
+                    missing_groups_and_fields['groups'].appends(path)
                 continue
             if not does_group_exist(renamed_path, data):
                 raise ValueError(f"The required group, {path}, hasn't been supplied.")
             continue
         if not is_path_in_data_dict or data[renamed_path] is None:
-            raise ValueError(f"The data entry corresponding to {path} is required "
-                             f"and hasn't been supplied by the reader.")
+            if entry_name.startswith('@'):
+                missing_groups_and_fields['attributes'].append(path)
+            else:
+                missing_groups_and_fields['fields'].append(path)
+
+    # Rasing errors for any kind of missing required fields, gorup and attributes:
+    def raise_required_concept_missing_error(missing_groups_and_fields):
+        error = ''
+        for concept, path_list in missing_groups_and_fields.items():
+            error = error + f"Required {concept} are not provided: \n"
+            if path_list:
+                for path in path_list:
+                    error = error + f"    \n\u2022{path}\n"
+        if error:
```

Comment:
```suggestion
        if missing_groups_and_fields:
```

Reply:
> 

It does not work, because `missing_group_and_fields` is not an empty dict.
```ipython
In [1]: dic = {'xx':[]}

In [2]: if dic:
   ...:     print('somthing')
   ...: 
somthing
```